src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -731,7 +732,12 @@ private static void CreatePart(ZipArchive package, IPackageFile file, Stream sou
             var entry = package.CreateEntry(entryName, CompressionLevel.Optimal);
             var physicalFile = file as PhysicalPackageFile;
             var lastWriteTime = physicalFile?.GetLastWriteTime();
-            if (lastWriteTime.HasValue) entry.LastWriteTime = lastWriteTime.Value;
+            if (lastWriteTime.HasValue)
+            {
+                // https://github.com/OctopusDeploy/OctopusClients/issues/245
+                // The minimum LastWriteTime for a ZIP entry is 1/1/1980
+                entry.LastWriteTime = new DateTime(Math.Max(lastWriteTime.Value.Ticks, MinZipLastWriteTime.Ticks));
```

Comment:
Instead of using Max and converting ticks, why not just do a comparison?

Reply:
`entry.LastWriteTime = lastWriteTime.Value < MinZipLastWriteTime ? MinZipLastWriteTime :  lastWriteTime.Value`
would be clearer